UnityProject/Assets/Scripts/Objects/Consoles/ArtifactConsole.cs

```diff
@@ -0,0 +1,135 @@
+using System.Collections;
+using System.Collections.Generic;
+using System;
+using UnityEngine;
+using Items.Science;
+using Objects.Research;
+using Mirror;
+using Shared.Systems.ObjectConnection;
+
+namespace Objects.Research
+{
+	public class ArtifactConsole : NetworkBehaviour, ICheckedInteractable<HandApply>, IMultitoolSlaveable
+	{
+		private RegisterObject registerObject;
+		public ItemStorage itemStorage;
+
+		public Artifact connectedArtifact;
+		public ArtifactDataDisk dataDisk = null;
+
+		public ArtifactData inputData = new ArtifactData();
+
+		[SyncVar] public bool HasDisk = false;
+
+		[SerializeField] ItemTrait ArtifactDiskTrait = null;
+
+		public Action StateChange;
+
+		private void Awake()
+		{
+			itemStorage = GetComponent<ItemStorage>();
+			registerObject = GetComponent<RegisterObject>();
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (interaction.HandSlot.IsEmpty) return false;
+			if (!DefaultWillInteract.Default(interaction, side)) return false;
+			if (Validations.HasItemTrait(interaction.UsedObject, ArtifactDiskTrait)) return true;
+
+			return false;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (itemStorage.GetIndexedItemSlot(0).IsEmpty)
+			{
+				Inventory.ServerTransfer(interaction.HandSlot, itemStorage.GetIndexedItemSlot(0));
+				dataDisk = itemStorage.GetIndexedItemSlot(0).ItemObject.GetComponent<ArtifactDataDisk>();
+				HasDisk = true;
+
+				Chat.AddActionMsgToChat(interaction.Performer, "You insert the drive into the console.",
+					interaction.Performer.ExpensiveName() + " inserts the dirve into the console.");
```

Comment:
I, too, insert my `dirve`.
```suggestion
					$"{interaction.Performer.ExpensiveName()} inserts the drive into the console.");
```